src/auxiliary/t8code.jl

```diff
@@ -14,30 +14,33 @@ function init_t8code()
             return nothing
         end
 
-        # Initialize the sc library, has to happen before we initialize t8code.
-        let catch_signals = 0, print_backtrace = 0, log_handler = C_NULL
-            T8code.Libt8.sc_init(mpi_comm(), catch_signals, print_backtrace, log_handler,
-                                 T8code.Libt8.SC_LP_ERROR)
+        # Initialize `libsc`, `p4est`, and `t8code` with log level
+        # `SC_LP_ERROR` to prevent a lot of output in AMR simulations
+        # For development, log level `SC_LP_DEBUG` is recommended.
+        LOG_LEVEL = T8code.Libt8.SC_LP_ERROR
+
+        if T8code.Libt8.sc_is_initialized() == 0
+            # Initialize the sc library, has to happen before we initialize t8code.
+            let catch_signals = 0, print_backtrace = 0, log_handler = C_NULL
+                T8code.Libt8.sc_init(mpi_comm(), catch_signals, print_backtrace,
+                                     log_handler,
+                                     LOG_LEVEL)
+            end
         end
 
         if T8code.Libt8.p4est_is_initialized() == 0
-            # Initialize `p4est` with log level ERROR to prevent a lot of output in AMR simulations
-            T8code.Libt8.p4est_init(C_NULL, T8code.Libt8.SC_LP_ERROR)
+            T8code.Libt8.p4est_init(C_NULL, LOG_LEVEL)
         end
 
-        # Initialize t8code with log level ERROR to prevent a lot of output in AMR simulations.
-        t8_init(T8code.Libt8.SC_LP_ERROR)
-
-        if haskey(ENV, "TRIXI_T8CODE_SC_FINALIZE")
```

Comment:
Why did you remove the effect of this environment variable?

Reply:
I decided that `sc_finalize` should be called always. A warning to the user is issued if there are still unfreed t8code objects lying around when Trixi shuts down. This is a serious issue since t8code allocates several MPI related objects (shared memory arrays). When they are not properly freed before MPI shuts down the program crashes before exiting in an orderly fashion.

See the code a couple of lines below.

Reply:
Due to the way the GC works in Julia I cannot influence the order of shutting down. Or at least, I couldn't find an elegant and robust way how to achieve this.

Reply:
Ok, thanks